pkg/stream/event_stream.go

```diff
@@ -0,0 +1,91 @@
+package stream
+
+import (
+	"context"
+	"crypto/tls"
+
+	"github.com/kelseyhightower/envconfig"
+	kafka "github.com/segmentio/kafka-go"
+	"github.com/segmentio/kafka-go/compress"
+	"github.com/segmentio/kafka-go/sasl/plain"
+)
+
+const EnableEventStreams = true
+const DisableEventStreams = false
+
+//go:generate mockgen -source=event_stream.go -package=stream -destination=mock_event_stream.go
+
+type KafkaProducer interface {
+	WriteMessages(ctx context.Context, msgs ...kafka.Message) error
+	Close() error
+}
+
+type EventEnvelope struct {
+	Name    string      `json:"name"`
+	Payload interface{} `json:"payload"`
+}
+
+type EventStreamWriter interface {
+	Write(ctx context.Context, key, value []byte) error
+	Close()
+}
+
+type KafkaConfig struct {
+	BootstrapServer string `envconfig:"KAFKA_BOOTSTRAP_SERVER" required:"true"`
+	ClientID        string `envconfig:"KAFKA_CLIENT_ID" default:""`
+	ClientSecret    string `envconfig:"KAFKA_CLIENT_SECRET" default:""`
+	Topic           string `envconfig:"KAFKA_EVENT_STREAM_TOPIC" required:"true"`
+}
+
+type KafkaWriter struct {
+	kafkaProducer KafkaProducer
+	kafkaTopic    string
+}
+
+func newProducer(config *KafkaConfig) KafkaProducer {
+
+	writer := &kafka.Writer{
+		Addr:        kafka.TCP(config.BootstrapServer),
+		Topic:       config.Topic,
+		Balancer:    &kafka.Hash{},
+		Compression: compress.Gzip,
+	}
+	if config.ClientID != "" && config.ClientSecret != "" {
+		mechanism := &plain.Mechanism{
+			Username: config.ClientID,
+			Password: config.ClientSecret,
+		}
+		writer.Transport = &kafka.Transport{
+			SASL: mechanism,
+			// let config pick default root CA, but define it to force TLS
+			TLS: &tls.Config{},
```

Comment:
yes, we are using TLS to transport this, however as the root CA is available in our underlying distribution there is no need to explicitly define which one to use, by just defining empty tls config it would pick it up from well-known system directories.
If we didn't want to encrypt, tls Config must be nil.